.github/workflows/license.yml

```diff
@@ -0,0 +1,70 @@
+name: License Check
+
+on: pull_request
+
+jobs:
+  license_job:
+    runs-on: ubuntu-latest
+    name: Run license checks on patch series (PR)
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v2
+      with:
+        path: ncs/nrf
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-doc-pip
+
+    - name: Install python dependencies
+      working-directory: ncs/nrf
+      run: |
+        pip3 install -U pip
+        pip3 install -U -r scripts/ci/requirements.txt
+        export PATH="$HOME/.local/bin:$PATH"
+        grep -E "west" scripts/requirements-fixed.txt | xargs pip3 install -U
+        pip3 show -f west
+
+    - name: West init and update
+      env:
+        BASE_REF: ${{ github.base_ref }}
+      working-directory: ncs/nrf
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        export PATH="$HOME/bin:$PATH"
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        git remote -v
+        git rebase origin/${BASE_REF}
+        # debug
+        git log  --pretty=oneline | head -n 10
+        west init -l .
+        west update
```

Comment:
can we skip west update?

Reply:
Not sure. I have to test how west will behave on initialized, but not updated workspace.

Reply:
if it isn't checking license on any files loaded with update, we can save 90 seconds.

Reply:
@doki-nordic @thst-nordic FYI, filed a bug here: https://github.com/zephyrproject-rtos/west/issues/588

Note that you don't need to do a full `west update` to get `west ncs-sbom`, you only need the `zephyr` project due to the `import` statement in the manifest file, so it is sufficient to do:
```
west update zephyr
```

Reply:
@mbolivar-nordic FYI, this is the use-case for https://github.com/zephyrproject-rtos/west/issues/588